### PR TITLE
add handling of deprecated test_code for `vt_scrape_test_code_editors`

### PR DESCRIPTION
**Pull request by mariev**

### CODE REVIEW

File: R/parse_roxygen.R [hunk 1]
```diff
@@ -161,25 +161,35 @@ parse_roxygen.r <- function(text){
...
   roxyblocks <- cleanup_section_last_update(roxyblocks)
 
-  roxyblocks <- lapply(roxyblocks,function(block){
+  roxyblocks <- lapply(roxyblocks,function(block, file){
+    test <- tryCatch({
+      as.list(block$call)[[2]]
+    }, error = function(e){
+
+      roxygen2::block_get_tag(block, "title")$val
+    })
```
Comment (thebioengineer) [inline R/parse_roxygen.R:173]: I thought we'd discussed flipping that? look for the title. if not, fall back on test description? 

The way test_that was written, it requires both a desc and the code, so I don't think the there can be a failure to fall back to title.

File: R/parse_roxygen.R [hunk 2]
```diff
@@ -161,25 +161,35 @@ parse_roxygen.r <- function(text){
...
+    }
 
     block$object <- structure(
       list(alias = test,
            topic = test,
-           value = block$call,
-           methods = NULL
+           value =  block$call,
+           methods = NULL,
+           file = file
```
Comment (thebioengineer) [inline R/parse_roxygen.R:184]: I like this addition! allows us to give more informative errors if necessary

File: tests/testthat/test-scrape_authors.R [hunk 3]
```diff
@@ -581,8 +581,50 @@ test_that("scrape roxygen tags from all sections authors", {
...
+  expect_warning(vt_scrape_test_code_editors(tags = c("editor", "editDate", "deprecate")),
+                 "No blocks with tags `editor`, `editDate`")
+
+  writeLines(c(
+    "#' @editor C user",
+    "#' @editDate 2001-01-1",
+    "#' @deprecate Deprecated in v1.2",
+    "NULL"),
+    con = vt_path("test_code","testcode1.R"))
+  expect_error(vt_scrape_test_code_editors(tags = c("editor", "editDate", "deprecate")))
```
Comment (thebioengineer) [inline tests/testthat/test-scrape_authors.R:612]: could you add the expected error text?

File: R/scrape_authors.R [hunk 4]
```diff
@@ -77,8 +77,15 @@ vt_scrape_tags_from <- function(type, tags = c("editor","editDate"), src = ".",
           }, block)
 
         item <- block$object$topic
+        if(is.null(item)){
 
-        do.call('data.frame', setNames(c(item, tag_values, FALSE), c(type, tags,"stringsAsFactors")))
+          abort(message = paste0("File ",
+                                  block$object$file,
+                                  " must include @title if deprecated."),
+                class = "vt.test_code_format")
```
Comment (thebioengineer) [inline R/scrape_authors.R:85]: Could we add a line/which test in the file? I think that would be more informative if possible.

File: R/parse_roxygen.R (hunk 1 above)
Comment (thebioengineer) [reply R/parse_roxygen.R:173]: I see now in the test the purpose now for this ordering. Maybe we should add more explicit testing to make sure that deprecated tests have the call NULL?

**Review by thebioengineer — CHANGES_REQUESTED** [review]

**Review by thebioengineer — APPROVED** [review]
